src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -178,26 +180,30 @@ public new CSharpParseOptions WithFeatures(IEnumerable<KeyValuePair<string, stri
                 throw new ArgumentNullException(nameof(features));
             }
 
-            // there are currently no parse options for experimental features
-            if (System.Linq.Enumerable.Any(features))
-            {
-                throw new ArgumentException("Experimental features are not supported", nameof(features));
-            }
+            var newFeatures = features.ToImmutableDictionary(kvp => kvp.Key, kvp => kvp.Value);
 
-            return this;
+            return new CSharpParseOptions(this) { _features = newFeatures };
         }
 
         public override IReadOnlyDictionary<string, string> Features
         {
             get
             {
-                // there are currently no parse options for experimental features
-                return new Dictionary<string, string>();
+                if (_features == null)
+                {
+                    _features = new Dictionary<string, string>();
```

Comment:
Use the following:

```
_features = ImmutableDictionary<string, string>.Empty
```

It avoids an allocation. 

Note: we could simplify this to just make `_features` default to this value.